apps/balancer-tools/src/components/Dialog.tsx

```diff
@@ -12,30 +12,39 @@ export function Dialog({
   subtitle,
   customWidth,
   noPadding = false,
+  isOpen,
+  setIsOpen,
+  onClose,
 }: React.PropsWithChildren<{
   content: React.ReactElement;
   title?: string;
   subtitle?: string;
   customWidth?: string;
   noPadding?: boolean;
+  isOpen?: boolean;
+  setIsOpen?: (isOpen: boolean) => void;
+  onClose?: (event: Event) => void;
 }>) {
   const [open, setOpen] = React.useState(false);
+  const isDialogOpen = isOpen ?? open;
+  const isDialogSetOpen = setIsOpen ?? setOpen;
   return (
-    <DialogPrimitive.Root open={open} onOpenChange={setOpen}>
+    <DialogPrimitive.Root open={isDialogOpen} onOpenChange={isDialogSetOpen}>
```

Comment:
🤔 Didn't completely get why we need this logic. Could you elaborate?

Reply:
Sure! By using const `[open, setOpen] = React.useState(false);` the dialog would be open on the `onClick` of the child component. But in the case of a Select, we want the dialog to be open only when the Select value changes. To achieve this I used an external state, but it's optional because if you don't provide the state the Dialog will open on `onClick` of the child

Reply:
so basically now we have two options to trigger the Dialog:
- Default: `onClick` of child component
- Custom: handle the open state outside of Dialog and use inside of it

Let me know if this was confusing 😅